src/backend/base/langflow/base/vectorstores/model.py

```diff
@@ -120,3 +145,11 @@ def get_retriever_kwargs(self):
         Get the retriever kwargs. Implementations can override this method to provide custom retriever kwargs.
         """
         return {}
+
+    @abstractmethod
+    @check_cached_vector_store
```

Comment:
It would have been nice if this actually enforced the decorator on all concrete implementations, but it doesn't (something to do with abstract methods?). So, subclasses have to remember to decorate this method in their implementation.

Reply:
```python
from abc import ABC, abstractmethod
import inspect

def check_cached_vector_store(func):
    def wrapper(*args, **kwargs):
        # Your decorator logic here
        return func(*args, **kwargs)
    return wrapper

class EnforcedDecoratorMeta(type):
    def __init__(cls, name, bases, dct):
        for name, value in dct.items():
            if hasattr(value, '__isabstractmethod__') and value.__isabstractmethod__:
                # If the method is abstract, ensure that it has @check_cached_vector_store in the subclass
                cls._check_method_decorator(name, cls)
        super().__init__(name, bases, dct)

    @staticmethod
    def _check_method_decorator(name, cls):
        method = getattr(cls, name)
        decorators = [dec.__name__ for dec in inspect.getmembers(method, predicate=inspect.isfunction)]
        if 'check_cached_vector_store' not in decorators:
            raise TypeError(f"Concrete implementation of '{name}' must use '@check_cached_vector_store' decorator.")

class LCVectorStoreComponent(Component, ABC, metaclass=EnforcedDecoratorMeta):

    @abstractmethod
    @check_cached_vector_store
    def get_retriever_kwargs(self):
        pass

class ConcreteVectorStore(BaseVectorStore):
    
    @check_cached_vector_store  # Must have this decorator or it will raise an error
    def get_retriever_kwargs(self):
        return {"example": "data"}
```

Something like this?

Reply:
Nice, that works